contracts/contracts/coordination/ManagedAllowList.sol

```diff
@@ -2,149 +2,102 @@
-     * @param _subscription The address of the subscription contract
      */
-    constructor(
-        Coordinator _coordinator,
-        UpfrontSubscriptionWithEncryptorsCap _subscription // TODO replace with IFeeModel subscription
-    ) GlobalAllowList(_coordinator) {
-        require(address(_subscription) != address(0), "Subscription cannot be the zero address");
-        subscription = _subscription;
+    constructor(Coordinator _coordinator) GlobalAllowList(_coordinator) {
+        _disableInitializers();
     }
 
-    /**
-     * @notice Checks if the sender is the authority of the ritual
-     * @param ritualId The ID of the ritual
-     */
-    modifier onlyCohortAuthority(uint32 ritualId) {
-        require(
-            coordinator.getAuthority(ritualId) == msg.sender,
-            "Only cohort authority is permitted"
-        );
-        _;
+    function ritualRole(uint32 ritualId, bytes32 role) public pure returns (bytes32) {
+        return keccak256(abi.encodePacked(ritualId, role));
     }
 
-    /**
-     * @notice Checks if the sender has allowance to set authorizations
-     * @dev This function overrides the canSetAuthorizations modifier in the GlobalAllowList contract
-     * @param ritualId The ID of the ritual
-     */
-    modifier canSetAuthorizations(uint32 ritualId) override {
-        require(getAllowance(ritualId, msg.sender) > 0, "Only administrator is permitted");
-        _;
+    function cohortAdminRole(uint32 ritualId) public pure returns (bytes32) {
+        return ritualRole(ritualId, COHORT_ADMIN_BASE);
     }
 
-    /**
-     * @notice Returns the allowance of an administrator for a ritual
-     * @param ritualId The ID of the ritual
-     * @param admin The address of the administrator
-     * @return The allowance of the administrator
-     */
-    function getAllowance(uint32 ritualId, address admin) public view returns (uint256) {
-        return allowance[LookupKey.lookupKey(ritualId, admin)];
+    function authAdminRole(uint32 ritualId) public pure returns (bytes32) {
+        return ritualRole(ritualId, AUTH_ADMIN_BASE);
     }
 
     /**
-     * @dev This function is called before the setAuthorizations function
+     * Acquire cohort admin role
      * @param ritualId The ID of the ritual
-     * @param addresses The addresses to be authorized
-     * @param value The authorization status
      */
-    function _beforeSetAuthorization(
-        uint32 ritualId,
-        address[] calldata addresses,
-        // TODO: Currently unused, remove?
-        bool value
-    ) internal override {
-        super._beforeSetAuthorization(ritualId, addresses, value);
-        for (uint256 i = 0; i < addresses.length; i++) {
-            require(
-                authActions[ritualId] <
-                    subscription.authorizationActionsCap(ritualId, addresses[i]),
-                "Authorization cap exceeded"
-            );
+    function initializeCohortAdminRole(uint32 ritualId) public {
```

Comment:
When it is expected this function to be called? The way I see it, we could do it in the coordinator when the ritual is initiated. This shouldn't matter in the case of ritual failing, but if ritual succeeds, then this is already done from day 1.

Reply:
I'd avoid any changes in Coordinator to call it 
This function included in grantAuthAdminRole function, so it will be called anyway during first granting